new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/TestAssertions.java

```diff
@@ -7,17 +7,109 @@
 
 import java.util.concurrent.Callable;
+import oracle.weblogic.kubernetes.assertions.impl.Kubernetes;
 
 // as in the actions, it is intended tests only use these assertaions and do
 // not go direct to the impl classes
 public class TestAssertions {
 
-    public static Callable<Boolean> operatorIsRunning() {
-        return Operator.isRunning();
-    }
+  /**
+   * Check if Operator is running
+   * @param namespace in which is operator is running
+   * @return true if running false otherwise
+   */
+  public static Callable<Boolean> operatorIsRunning(String namespace) {
+    return Operator.isRunning(namespace);
+  }
 
-    public static Callable<Boolean> domainExists(String domainUID, String namespace) {
-        return Domain.exists(domainUID, namespace);
-    }
+  /**
+   * check if operator Rest service is running
+   * @param namespace the operator rest service exists
+   * @return true if rest service is running otherwise false
+   */
+  public static boolean operatorRestServiceRunning(String namespace) {
+    return Operator.isRestServiceCreated(namespace);
+  }
+
+  /**
+   * check if a weblogic custom resource domain exists in specified namespace and all its pods are running
+   * @param domainUID ID of the domain
+   * @param namespace in which the domain custom resource exists
+   * @return true if domain exists and pods running otherwise false
+   */
+  public static Callable<Boolean> domainExists(String domainUID, String namespace) {
+    return Domain.exists(domainUID, namespace);
+  }
+
+  /**
+   * check if a kubernetes pod is in running/ready state
+   * @param podName name of the pod to check for
+   * @param domainUID weblogic domain uid in which the pod belongs
+   * @param namespace in which the pod is running
+   * @return true if the pod is running otherwise false
+   */
+  public static Callable<Boolean> podReady(String podName, String domainUID, String namespace) {
+    return Kubernetes.podRunning(podName, domainUID, namespace);
+  }
+
+  /**
+   * check if a pod given by the podName is in Terminating state
+   * @param podName name of the pod to check for Terminating status
+   * @param domainUID weblogic domain uid in which the pod belongs
+   * @param namespace in which the pod is running
+   * @return true if the pod is terminating otherwise false
+   */
+  public static Callable<Boolean> podTerminating(String podName, String domainUID, String namespace) {
+    return Kubernetes.podTerminating(podName, domainUID, namespace);
+  }
+
+  /**
+   * check is a service exists in given namespace
+   * @param serviceName  the name of the service to check for
+   * @param namespace in which the service is running
+   * @return true if the service exists otherwise false
+   */
+  public static boolean serviceReady(String serviceName, String namespace) {
+    return Kubernetes.serviceCreated(serviceName, namespace);
+  }
+
+  /**
+   * check if a loadbalancer pod is ready
```

Comment:
Sentences start with a capital letter and end with a period.

Reply:
fixed